src/data/dynamo/table.ts

```diff
@@ -1,191 +1,331 @@
+          )
+          reject(err)
+        } else {
+          const models = documents
+            .map(document => this.mapFromDynamoItem(document.attrs))
+            .map(document => Model.createModel(this.TConstructor, document))
+
+          resolve(models)
+        }
+      }
+
+      this.dynModel.getItems(items, options, callback)
     }
 
-    public getAllRaw(): Promise<{}[]> {
-        return this.getAllBase(x => x);
+    return new Promise(promiseCallback)
+  }
+
+  public get(
+    hashKey: string,
+    options: GetItemOptions = {},
+    rangeKey?: string
+  ): Promise<TModel> {
+    let promiseCallback = (resolve: Function, reject: Function) => {
+      let callback: DynogelsItemCallback = (err, document) => {
+        if (err) {
+          console.error(
+            `Error getting items on ${this.tableName} table. Err: ${err}`
+          )
+          reject(err)
+        } else if (document === undefined || document === null) {
+          resolve(undefined)
+        } else {
+          const model = Model.createModel(
+            this.TConstructor,
+            this.mapFromDynamoItem((document as any).attrs)
+          )
+          resolve(model)
+        }
+      }
+
+      if (rangeKey == null) {
+        this.dynModel.get(hashKey, options, callback)
+      } else {
+        this.dynModel.get(hashKey, rangeKey, options, callback)
+      }
     }
 
-    public getItems(items: any[], options: GetItemOptions = {}): Promise<TModel[]> {
-
-        let promiseCallback = (resolve, reject) => {
-            let callback = (err, documents: any[]) => {
-                if (err) {
-                    console.error(`Error getting items on ${this.tableName} table. Err: ${err}`);
-                    reject(err);
-                }
-                else {
-                    const models = documents.map(document => new this.TConstructor(document.attrs));
-                    resolve(models);
-                }
-            };
-
-            this.dynModel.getItems(items, options, callback);
-        };
+    return new Promise(promiseCallback)
+  }
+
+  public getOrFail(
+    hashKey: string,
+    options: GetItemOptions = {},
+    rangeKey?: string
+  ): Promise<TModel> {
+    return this.get(hashKey, options, rangeKey).then(model => {
+      if (model === undefined) {
+        const keyMsg =
+          `HashKey: ${hashKey} ` + (rangeKey === undefined)
+            ? ''
+            : `, RangeKey: ${rangeKey}`
+
+        console.error(
+          `No item with ${keyMsg} found on ${this.tableName} table.`
+        )
+        return Promise.reject('Model not found')
+      }
+      return Promise.resolve(model)
+    })
+  }
+
+  public update(data: TModel, options: UpdateItemOptions = {}): Promise<void> {
+    return this.updateRaw(data.model, options)
+  }
+
+  public updateRaw(
+    data: { [key: string]: {} },
+    options: UpdateItemOptions = {}
+  ): Promise<void> {
+    const mappedItem = this.mapToDynamoItem(data)
```

Comment:
Can we move this into `update`? I feel like update raw shouldn't do any mappings

Reply:
im on the fence for this. I feel like you would still want the attributes mapped. It has to be mapped at some point.
If you range key is height:weight then you going to have to set that as the range key at some point I think we still want to do that so they can just pass us {height: blah, weight: blah} instead of handling mapping on their end. Thoughts?

Reply:
when I think of `updateRaw` I think it should just be updating with what I give it any nothing else. I'm sure there are use cases where people would rather put in:

```ts
{
   "dimensions": "10:10"
}
```

than

```ts
{
   "height":10,
   "weight":10
}
```
I think that `raw` should be reserved for doing direct updates to an item with no extra bs.